### PR TITLE
Clearing tags

### PR DESCRIPTION
**Pull request by BartSoftware**

### CODE REVIEW

File: app/src/main/java/com/invisibleteam/goinvisible/mvvm/edition/TagsManager.java [hunk 1]
```diff
@@ -0,0 +1,292 @@
...
+            TAG_GPS_DATESTAMP,
+            TAG_GPS_LATITUDE_REF,
+            TAG_GPS_TIMESTAMP,
+            TAG_GPS_LONGITUDE_REF,
+            TAG_GPS_PROCESSING_METHOD,
+            TAG_DATETIME,
+            TAG_MAKE,
+            TAG_MODEL);
+
+    private static final List<String> STRING_TAGS_LIST_API_24 = Arrays.asList(
```
Comment (kejmil01) [inline app/src/main/java/com/invisibleteam/goinvisible/mvvm/edition/TagsManager.java:111]: Lint annotation missing

File: app/src/main/java/com/invisibleteam/goinvisible/mvvm/edition/TagsManager.java [hunk 2]
```diff
@@ -0,0 +1,292 @@
...
+
+    private static final List<String> INTEGER_TAGS_LIST = Arrays.asList(
+            TAG_IMAGE_LENGTH,
+            TAG_WHITE_BALANCE,
+            TAG_GPS_ALTITUDE_REF,
+            TAG_FLASH,
+            TAG_IMAGE_WIDTH,
+            TAG_ORIENTATION);
+
+    private static final List<String> INTEGER_TAGS_LIST_API_24 = Arrays.asList(
```
Comment (kejmil01) [inline app/src/main/java/com/invisibleteam/goinvisible/mvvm/edition/TagsManager.java:158]: Lint annotation missing

File: app/src/main/java/com/invisibleteam/goinvisible/mvvm/edition/TagsManager.java [hunk 3]
```diff
@@ -0,0 +1,292 @@
...
+            TAG_SUBJECT_DISTANCE_RANGE,
+            TAG_SUBJECT_LOCATION,
+            TAG_GPS_DIFFERENTIAL,
+            TAG_THUMBNAIL_IMAGE_LENGTH,
+            TAG_THUMBNAIL_IMAGE_WIDTH);
+
+    private static final List<String> DOUBLE_TAGS_LIST = Arrays.asList(
+            TAG_EXPOSURE_TIME);
+
+    private static final List<String> DOUBLE_TAGS_LIST_API_24 = Arrays.asList(
```
Comment (kejmil01) [inline app/src/main/java/com/invisibleteam/goinvisible/mvvm/edition/TagsManager.java:201]: Lint annotation missing

File: app/src/main/res/layout/edit_item_view.xml [hunk 4]
```diff
@@ -1,45 +1,66 @@
...
+                android:layout_toEndOf="@+id/tag_key"
+                android:layout_toStartOf="@+id/erase_button"
+                android:maxLines="2"
                 android:text="@{viewModel.value}"
                 android:textColor="@android:color/black"
-                android:textSize="16sp" />
+                android:textSize="16sp"/>
+
+            <ImageView
+                android:id="@+id/erase_button"
```
Comment (kejmil01) [inline app/src/main/res/layout/edit_item_view.xml]: I think it should be "clear_button"

File: app/src/test/java/com/invisibleteam/goinvisible/mvvm/edition/TagsManagerTest.java [hunk 5]
```diff
@@ -0,0 +1,71 @@
...
+
+        //When
+        List<Tag> tagList = provider.getAllTags();
+
+        //Then
+        assertThat(tagList.get(0), equalsTag(new Tag(TAG_MAKE, "tag1", ObjectType.STRING)));
+        assertThat(tagList.get(1), equalsTag(new Tag(TAG_MODEL, "tag2", ObjectType.STRING)));
+    }
+
+//    @Test
```
Comment (kejmil01) [inline app/src/test/java/com/invisibleteam/goinvisible/mvvm/edition/TagsManagerTest.java]: I would remove that comment block

File: app/src/test/java/com/invisibleteam/goinvisible/mvvm/edition/TagsManagerTest.java [hunk 6]
```diff
@@ -0,0 +1,71 @@
...
+import static org.junit.Assert.assertThat;
+import static org.mockito.Mockito.mock;
+import static org.mockito.Mockito.when;
+
+@RunWith(RobolectricTestRunner.class)
+@Config(constants = BuildConfig.class, sdk = 21)
+public class TagsManagerTest {
+
+    @Test
+    public void whenThereIsNoImageAtPath_ReturnEmptyList() throws IOException {
```
Comment (kejmil01) [inline app/src/test/java/com/invisibleteam/goinvisible/mvvm/edition/TagsManagerTest.java]: Name of this test doesn't match with body

File: app/src/main/java/com/invisibleteam/goinvisible/mvvm/edition/adapter/EditItemAdapter.java [hunk 7]
```diff
@@ -27,8 +30,11 @@ public ViewHolder onCreateViewHolder(ViewGroup parent, int viewType) {
 
     @Override
     public void onBindViewHolder(ViewHolder holder, int position) {
-        holder.itemView.setTag(tagsList.get(position));
-        holder.editItemViewModel.setModel(tagsList.get(position));
+        final Tag tag = tagsList.get(position);
+
+        holder.itemView.setTag(tag);
+        holder.editItemViewModel.setModel(tag);
+        holder.editItemViewBinding.eraseButton.setOnClickListener(v -> onTagActionListener.onClear(tag));
```
Comment (kejmil01) [inline app/src/main/java/com/invisibleteam/goinvisible/mvvm/edition/adapter/EditItemAdapter.java]: I would like to see TODO comment, smth like "think about moving it to onCreateViewHolder in the future"

File: app/src/main/java/com/invisibleteam/goinvisible/mvvm/edition/adapter/EditItemAdapter.java [hunk 8]
```diff
@@ -40,10 +46,27 @@ void updateImageList(List<Tag> imageList) {
         this.tagsList = imageList;
     }
 
+    void setOnTagActionListener(OnTagActionListener listener) {
+        this.onTagActionListener = listener;
+    }
+
+    void clearTag(Tag tag) {
+        if (!tagsList.isEmpty()) {
+            for (int index = 0; index < tagsList.size(); index++) {
```
Comment (kejmil01) [inline app/src/main/java/com/invisibleteam/goinvisible/mvvm/edition/adapter/EditItemAdapter.java:56]: Maybe we should store models in some kind of map to  have easier and quicker lookups

File: app/src/test/java/com/invisibleteam/goinvisible/mvvm/edition/TagsManagerTest.java (hunk 5 above)
Comment (michal1990) [reply app/src/test/java/com/invisibleteam/goinvisible/mvvm/edition/TagsManagerTest.java]: or @Ignore it, in case when You would keep this test

**Review by kejmil01 — APPROVED** [review]

**Pull request merged**